### DISCUSSION TITLE
How to use getFetch in onMounted?

### DISCUSSION THREAD

**Discussion by speg** [nuxt/nuxt#16466] [Questions]
Hi all,

I just came back full of Kool-Aid from VueConf Toronto and started to write our Nuxt2 / Vue2 app into the latest and greatest! 😝 

TypeScript, Nuxt3, Vue3, Pinia, Vuetify3, this is going to be amazing! I'm laying down the pieces and everything is going swimmingly. This is going to be great!

 So I eventually get to the point where I want to load some data from the network `server/api` works great! Next I try and use  `useFetch` from Nuxt and _something_ goes awry.

Without thinking too much about it; I drop an `await useFetch(..)` into `onMounted` and ... nothing? Oh wait - once in a while it works. Weird. Its not even making the network request. Play around with some options like`initialCache` and whatnot without any changes.

Then I decide `async` inside `onMounted` isn't a good idea (I thought I read that somewhere, but it was getting late and my eyes may deceive me) and so I swap `onMounted` for `onBeforeMount` and suddenly everything works. Every time.

I'm not working in the frontend full time so I may be missing something obvious. Misusing Vue, or Nuxt. Let me know if you can spot my mistake! 🙏🏼 

```vue
<script setup lang="ts">
  const props = defineProps(['resource'])
  const resource = ref(null)  

  // onMounted(async () => {          // <-- doesn't work, most of the time
  onBeforeMount(async () => {         // <-- works all the time every time!
    if (props.resource) {
      const { data: value } = await useFetch(props.resource)
      resource.value = value
    } 
  })
</script>

<template>
  <v-card>
    <slot />
    <pre>{{ resource }}</pre>
  </v-card>
</template>


```

**Comment by wobsoriano** [comment]
You should not be putting those inside lifecycle hooks. There’s a server: false option if you want it to run on client only.
